encoding/codecv7.go

```diff
@@ -121,23 +122,28 @@ func (d *DACodecV7) constructBlob(batch *Batch) (*kzg4844.Blob, common.Hash, []b
 
 	if len(blobBytes) > maxEffectiveBlobBytes {
 		log.Error("ConstructBlob: Blob payload exceeds maximum size", "size", len(blobBytes), "blobBytes", hex.EncodeToString(blobBytes))
-		return nil, common.Hash{}, nil, fmt.Errorf("blob exceeds maximum size: got %d, allowed %d", len(blobBytes), maxEffectiveBlobBytes)
+		return nil, common.Hash{}, nil, common.Hash{}, fmt.Errorf("blob exceeds maximum size: got %d, allowed %d", len(blobBytes), maxEffectiveBlobBytes)
 	}
 
 	// convert raw data to BLSFieldElements
 	blob, err := makeBlobCanonical(blobBytes)
 	if err != nil {
-		return nil, common.Hash{}, nil, fmt.Errorf("failed to convert blobBytes to canonical form: %w", err)
+		return nil, common.Hash{}, nil, common.Hash{}, fmt.Errorf("failed to convert blobBytes to canonical form: %w", err)
 	}
 
 	// compute blob versioned hash
 	c, err := kzg4844.BlobToCommitment(blob)
 	if err != nil {
-		return nil, common.Hash{}, nil, fmt.Errorf("failed to create blob commitment: %w", err)
+		return nil, common.Hash{}, nil, common.Hash{}, fmt.Errorf("failed to create blob commitment: %w", err)
 	}
 	blobVersionedHash := kzg4844.CalcBlobHashV1(sha256.New(), &c)
 
-	return blob, blobVersionedHash, blobBytes, nil
+	paddedBlobBytes := make([]byte, maxEffectiveBlobBytes)
+	copy(paddedBlobBytes, blobBytes)
```

Comment:
Why copying here? Also wouldn't we need to call `paddedBlobBytes = makeBlobCanonical(blobBytes)` again?

Why not just use `blob` here? `blob` is padded and correct size

Reply:
`paddedBlobBytes = makeBlobCanonical(blobBytes)` this is after 31 bytes per 32 bytes conversion.

Reply:
comments are added in: [5f1aec7](https://github.com/scroll-tech/da-codec/pull/50/commits/5f1aec7a8aee898843a0ca785cf8cffca6d5b097)